### ISSUE TITLE
Multiple Different Errors Upon Reserialization of Configured Items

### ISSUE THREAD

**Issue by MrSnappingTurtle** [kamsar/Unicorn#45]
For seemingly no reason, I am getting two different kinds of errors upon trying to "Reserialize Default Configuration Now" in the control panel. 

The first error is like the following, and seems to change the path that it throws at random:

```
ERROR: Access to the path 'Random Path in the serialization folder' is denied. (System.UnauthorizedAccessException)
at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath) 
at System.IO.Directory.DeleteHelper(String fullPath, String userPath, Boolean recursive, Boolean throwOnTopLevelDirectoryNotFound) 
at System.IO.Directory.Delete(String fullPath, String userPath, Boolean recursive, Boolean checkHost) 
at Unicorn.Serialization.Sitecore.SitecoreSerializationProvider.DeleteItemRecursive(ISerializedReference reference) 
at Unicorn.Serialization.Sitecore.SitecoreSerializationProvider.DeleteItemRecursive(ISerializedReference reference) 
at Unicorn.Serialization.Sitecore.SitecoreSerializationProvider.DeleteItemRecursive(ISerializedReference reference) 
at Unicorn.Serialization.Sitecore.SitecoreSerializationProvider.DeleteItemRecursive(ISerializedReference reference) 
at Unicorn.Serialization.Sitecore.SitecoreSerializationProvider.DeleteSerializedItem(ISerializedReference item) 
at Unicorn.ControlPanel.ReserializeConsole.Process(IProgressStatus progress)
```

The second error is consistent and happens when it reaches sitecore/layout and hasn't already crashed from the previous error:

```
ERROR: Object reference not set to an instance of an object. (System.NullReferenceException)
at Sitecore.Data.Serialization.ItemSynchronization.BuildSyncItem(Item item) 
at Unicorn.Serialization.Sitecore.SitecoreSerializationProvider.SerializeItem(ISourceItem item) 
at Unicorn.ControlPanel.ReserializeConsole.Serialize(ISourceItem root, IPredicate predicate, ISerializationProvider serializationProvider, ILogger logger) 
at Unicorn.ControlPanel.ReserializeConsole.Serialize(ISourceItem root, IPredicate predicate, ISerializationProvider serializationProvider, ILogger logger) 
at Unicorn.ControlPanel.ReserializeConsole.Serialize(ISourceItem root, IPredicate predicate, ISerializationProvider serializationProvider, ILogger logger) 
at Unicorn.ControlPanel.ReserializeConsole.Process(IProgressStatus progress)
```

I have tried reinstalling Unicorn multiple times. I get the same errors as above if I first delete the serialization folder and try to do the initial setup in the control panel again.

**Comment by demisx** [comment]
+1 Same issue.

**Comment by kamsar** [comment]
What Sitecore version is this on?

**Comment by demisx** [comment]
@kamsar 7.5

**Comment by kamsar** [comment]
Hmm, the first error might be caused by an antivirus or other filesystem-watching program locking files as Unicorn messes with them. I have not seen it before; but then I also have not tested it with 7.5 (I skipped directly to 8). Might try deleting the item tree in windows explorer, so that Unicorn doesn't have to clear them out before reserializing.

As to the second error, the stack trace jumps down into Sitecore's serialization API. The stack within Unicorn is pretty defensive (e.g. we're asserting that we're not passing any nulls to `BuildSyncItem`). Seems like it might be a bug in 7.5's serialization APIs not present in 8.x, or there might be something unique about the item you're trying to serialize. (That said I have a site where I keep the whole master and core stock DBs under Unicorn, so it sees a lot of unique things)

**Comment by MrSnappingTurtle** [comment]
@kamsar Hi, we ended up fixing the issues and I think that they might have been initially caused by some problematic items in our database. We had initially tried serializing using the default `<include />` paths. We had a massive database and, as I later realized, some items that were possibly corrupt. These items claimed to have children and if you were to try opening them it would say they didn't exist. Trying to package or serialize them caused errors. I think that upon trying to serialize these items, which were included in the default configuration we used, Unicorn started to stumble into issues. I think this also may have corrupted our database somehow as the errors started appearing after that. 

We recently reset the database but this time limited the `<include />` paths to only items that we were adding/modifying. It now works without issue. 

Thanks.
